[Lee Su Kiat Sean] iP

sean6369 opened the pull request.

sean6369 committed:
Rename bot to Finch, Greet, Exit

sean6369 committed:
Add echo of commands

Echos commands entered by the user
Exits when the user types the command bye
Add additional ASCII art for bot name

sean6369 committed:
Add add and list

Add the ability to store whatever text entered by the user
And the ability to display them back when user enters "list"
Add indentation for bot replies

sean6369 committed:
Add Mark as Done fuctionality

Add task class to manage task
Users can mark tasks as done or change the status to not done

sean6369 committed:
Tweak the code to comply with coding standards

shauntsr commented on src/main/java/Finch.java:
```diff
@@ -0,0 +1,106 @@
+    // Constant for the bot's name
+    private static final String NAME = "Finch";
+
+    // Prints a horizontal line
+    private static void line() {
+        System.out.println("    ____________________________________________________________");
+    }
```
Perhaps add a verb to the naming ie printHorizontalLine?

shauntsr commented on src/main/java/Finch.java:
```diff
@@ -0,0 +1,106 @@
+            } else {
+                addTask(userInput);
+            }
+        }
```
Perhaps a switch case could make this cleaner?

shauntsr commented on src/main/java/Finch.java:
```diff
@@ -0,0 +1,106 @@
+public class Finch {
+
+    // Constant for the bot's name
+    private static final String NAME = "Finch";
```
I like the use of a constant to store the chatbot's name.

shauntsr commented on src/main/java/Finch.java:
```diff
@@ -0,0 +1,106 @@
+            line();
+        }
+    }
+}
```
I like the consistency in using camelCase for your variables and methods.

shauntsr reviewed (commented):
LGTM. The code was very clean and easy to read.

Yxiang-828 commented on src/main/java/Finch.java:
```diff
@@ -71,7 +71,7 @@ private static void addTask(String taskDescription) {
         System.out.println("    Here are your tasks in your list:");
         for (int i = 0; i < taskCount; i++) {
-            System.out.println("    "  + (i + 1) + ". " + tasks[i]);
+            System.out.println("    " + (i + 1) + ". " + tasks[i]);
```
4 space indentation, good

Yxiang-828 commented on src/main/java/Finch.java:
```diff
@@ -0,0 +1,19 @@
+public class Finch {
```
I reckon a javadoc for each major segment, such as here you are missing a class level javadoc:
/**
 * The main class for the Finch...
 * Does...
 */

Yxiang-828 commented on src/main/java/Finch.java:
```diff
@@ -7,9 +7,15 @@ public class Finch {
 
     // Prints a horizontal line
     private static void line() {
```
naming could be improved based on coding standard, such as printLine may be better as verbs are more understandable

Yxiang-828 commented on src/main/java/Finch.java:
```diff
@@ -65,10 +69,38 @@ private static void addTask(String task) {
+            System.out.println("    " + tasks[taskIndex]);
+            line();
+        } else {
+            System.out.println("    Task not found.");
```
nice, clear print statement to tell user that they should read instructions properly and do again

sean6369 committed:
Add ToDos, Events, Deadlines

Add new class Todo, Deadline and Event that inherits from Task class to handle more commands
Changed if-else statements to switch cases for better readability

sean6369 committed:
Automated UI Text Testing

Created test input and expected output files for automated testing
Run runtest.sh to compare the difference and check if the test passes or fails

sean6369 committed:
add error handling for unknown commands

able to gracefully handle empty inputs, invalid commands and any other unexpected errors

sean6369 committed:
Add error handling for mark and unmark tasks

sean6369 committed:
Add error handling for list method

sean6369 committed:
Add error handling to addToDo and addTask method

sean6369 committed:
Add error handling for addDeadline method

sean6369 committed:
Add error handling for addEvent method

Add extra error handling for mark and unmark, for if user inputs more than 1 task number
Remove try-catch statement for addDeadline method to be consistent with addEvent method

sean6369 committed:
Merge branch 'branch-Level-5'

sean6369 committed:
Add FinchException class and updated mark and unmark methods

FinchException class handles chatbot-specific error messages
This improves error handling by providing clearer, consistent messages instead of using generic Java exceptions

sean6369 committed:
Update addTask, add ToDo and list methods to throw FinchException for error handling

sean6369 committed:
update addDeadline and addEvent methods to throw FinchException for error handling

sean6369 committed:
Refactor addDeadline and addEvent methods into helper methods parseDeadline and parseEvent respectively

This makes for a cleaner and shorter addDeadline and addEvent method

sean6369 committed:
Merge branch 'branch-Level-5'

sean6369 committed:
Organize into packages

Create 3 main packages, finch.ui, finch.task and finch.exception

sean6369 committed:
Update the runtest.sh file directory

Updated the EXPECTED.TXT to match the changes made from error handling

sean6369 committed:
Merge branch 'branch-A-Packages'

namsengi11 commented on src/main/java/finch/ui/Finch.java:
```diff
@@ -0,0 +1,287 @@
+public class Finch {
+
+    // Constant for the bot's name
+    private static final String NAME = "Finch";
```
Good job using uppercase for constants

namsengi11 commented on src/main/java/finch/ui/Finch.java:
```diff
@@ -0,0 +1,287 @@
+    }
+
+    // Array to store tasks
+    private static final Task[] tasks = new Task[100];
```
Could factor 100 out as a new variable, such as MAX_TASKS_LENGTH

namsengi11 commented on src/main/java/finch/ui/Finch.java:
```diff
@@ -0,0 +1,287 @@
+                        ██╔══╝  ██║██║╚██╗██║██║     ██╔══██║
+                        ██║     ██║██║ ╚████║╚██████╗██║  ██║
+                        ╚═╝     ╚═╝╚═╝  ╚═══╝ ╚═════╝╚═╝  ╚═╝
+                """.stripTrailing());
```
Cool text art!

namsengi11 commented on src/main/java/finch/ui/Finch.java:
```diff
@@ -0,0 +1,287 @@
+                    continue;
+                }
+                printHorizontalLine();
+                String[] inputParts = userInput.trim().split("\\s+"); // Splits the command by spaces (multiple spaces handled as well)
```
Good job commenting on what regex intends to achieve

sean6369 committed:
Add delete command

sean6369 committed:
Add save function

List of tasks are saved automatically in the finch.txt file, located in the data folder when the task list changes
Tasks are loaded from the hard-disk when the chatbot starts up

sean6369 committed:
Merge branch 'branch-Level-6'

sean6369 committed:
Merge branch 'branch-Level-7'

# Conflicts:
#	src/main/java/finch/ui/Finch.java

Add save to storage for delete command
Change data structure from fixed array to arrayList

sean6369 committed:
Package the app as a JAR file

sean6369 committed:
Add more usage of OOP

New Ui class, TaskList class and Parser class
Update Storage class and other relevant command methods to utilise new classes
Finch class now under finch package and updated to refactored code

sean6369 committed:
Add even more usage of OOP

Refactored into individual command classes, that inherits from an abstract Command class

sean6369 committed:
Add the chatbot to understand dates

Deadline and Event command take in dates in the YYYY-MM-DD format

sean6369 committed:
Add chatbot to understand time

sean6369 committed:
Code cleanup

Removed unused import statements
Edit deadline and event class to use 1 constructor instead of 2

sean6369 committed:
Add new command find

Allows user to find tasks by searching for keywords in the description of tasks

sean6369 committed:
Merge pull request #1 from sean6369/branch-Level-9

Add new command find

sean6369 committed:
Add JavaDocs comments to the code

sean6369 committed:
Merge pull request #2 from sean6369/branch-A-JavaDoc

Add JavaDocs comments to the code

sean6369 committed:
Updated readme to show a user guide

sean6369 committed:
Release Jar file

sean6369 committed:
Add more javadocs

Removed some unused import statements
Added "find" command in the list of available commands

sean6369 committed:
Updated the correct readme file under the docs folder

sean6369 committed:
Clean up code

Removed redundant or duplicate code
